src/connection/manager.js

```diff
@@ -78,14 +79,26 @@ class ConnectionManager {
    * @returns {void}
    */
   remove (connection) {
-    if (!this.connections[connection.theirB58Id]) return
+    // No record of the peer, disconnect it
+    if (!this.connections[connection.theirB58Id]) {
+      connection.theirPeerInfo.disconnect()
+      this.switch.emit('peer-mux-closed', connection.theirPeerInfo)
+      return
+    }
 
     for (let i = 0; i < this.connections[connection.theirB58Id].length; i++) {
```

Comment:
I feel like the code in this class could be simpler with `Set` and `Map` - suggestion for a future PR

Reply:
I agree, a `Map` would allow for handling the two dimensional array pretty cleanly.